src/render.js

```diff
@@ -1,48 +1,40 @@
-import entries from 'object.entries'
-
-const {hasOwnProperty} = Object.prototype
-const tags = {}
-let prevStyles = {}
+const tags = new Map()
+let prevStyles = new Map()
 
 export default typeof window === 'undefined' ? renderOnServer : renderOnClient
 
 function renderOnServer() {}
 
-function renderOnClient(components) {
-  const styles = {}
-  for (const c of components) {
-    styles[c.props.styleId] = c
```

Comment:
I think we can do `if (!styles.hasOwnProperty(c.props.styleId))` check to fix the issue instead.
tho I think it would be ok if we decided to require `Map`.